CONTRIBUTING.txt

```diff
@@ -0,0 +1,98 @@
+    7. A PR SHALL NOT include non-trivial code from other projects unless the Contributor is the original author of that code.
+    8. A PR MUST compile cleanly and pass project self-tests on all target platforms.
+    9. Each commit message in a PR MUST succinctly explain what the commit achieves.
+    10. The PR Description MUST consist of a single short (less than 50 characters) line stating the problem (“Problem: …") being solved, followed by a blank line and then the proposed solution (“Solution: …").
+    11. A “Valid PR” is one which satisfies the above requirements.
+
+6. Process
+
+    1. Change on the project follows the pattern of accurately identifying problems and applying minimal, accurate solutions to these problems.
+    2. To request changes, a user (end user, contributor, maintainer, or administrator) logs an issue on the project Platform issue tracker.
+    3. The user or contributor should write the issue by describing the problem they face or observe. Links to the forum or other resources are permitted but the issue should be complete and accurate and should not require the reader to visit the forum or any other platform to understand what is being described.
+    4. Issues should strive to describe the minimum acceptable condition.
+    5. Issues should focus on user tasks and avoid comparisons to other software solutions.
+    6. The user or Contributor SHOULD seek consensus on the accuracy of their observation and the value of solving the problem.
+    7. To submit a solution to a problem, a Contributor SHALL create a pull request back to the project.
+    8. Contributors and Maintainers SHALL NOT commit changes directly to the target branch.
+    9. To discuss a proposed solution, users MAY comment on the platform pull request. Forum conversations regarding the solution should be discouraged and conversation redirected to the Pull Request or the related issue.
+    10. To accept or reject a Pull Request, a Maintainer SHALL use the platform interface.
+    11. Maintainers SHOULD NOT merge their own PRs except:
+      - in exceptional cases, such as non-responsiveness from other Maintainers for an extended period (more than 1-2 days).
+      - If the maintainer is also the primary developer of the workbench or subsystem.
+
+    12. Maintainers SHALL merge valid PRs from other Contributors rapidly.
+    13. Maintainers MAY, at their discretion merge PRs that have not met all criteria to be considered valid to:
+        - (a) ending fruitless discussions
+        - (b) capturing toxic patches in the historical record
+        - (c) engaging with the Contributor on improving their patch quality.
+    14. Maintainers SHALL NOT make value judgments on correct patches.
+    15. Any Contributor who has value judgments on a PR SHOULD express these via their own PR.
+    16. The user who created an issue SHOULD close the issue after checking the PR is successful.
```

Comment:
Why not use the git `Closes: ` tag together with 'MUST close'? The suggested way, adds additional workload for the maintainers, which could be avoided by automatic tasks on the platform.
Suggestion: The User who created an issue MUST close the issue using a git 'Closes: ' tag in the PR.

Reply:
I believe that's a Github tag rather than a Git tag.

Reply:
Maybe replace with "The User who created an issue (or in their absence, a Maintainer)..."

Reply:
```suggestion
    16. The User who created an issue SHOULD close the issue after checking the PR is successful.
```

Reply:
> I believe that's a Github tag rather than a Git tag.

AFAIK it's git related, see for example https://www.gentoo.org/glep/glep-0066.html#commit-messages. But no matter where it's rooted, my point is about automation and reducing the possible work of people working on the PR.

Reply:
Thanks. I did not know that. Not sure if a commit tag should automatically close issues. For example, maybe the Contributor added the tag but the problem is not totally closed, and the Contributor isn't removing the tag for whatever reason.

Reply:
Until the problem isn't totally solved, the PR usually wouldn't get merged. The issue will only be closed on merging, not when creating the PR. If a PR is only a partial solution to a problem, the Closes: tag shouldn't be used, but instead the Bug: tag could be used.
On Gentoo, we work a lot with these tags and from my experience, they work very well.
But it's only a suggestion.

Reply:
> Suggestion: The User who created an issue MUST close the issue using a git 'Closes: ' tag in the PR.

It's a SHOULD for practical reasons. What will we do in the case where the user disappears forever?

Reply:
The User who created the issue isn't the one who created the PR (in the general case). In an ideal case, the user who created the issue verifies that the user who created the fix did it correctly, as a double-check.

Reply:
> > Suggestion: The User who created an issue MUST close the issue using a git 'Closes: ' tag in the PR.
> 
> It's a SHOULD for practical reasons. What will we do in the case where the user disappears forever?

That's only a problem, if the user has shut down it's github account. If the user is just not responsive it depends on the quality of the PR. If it can be merged without the need for changes, all is fine, if changes are needed we have a problem. But I see your point.

Reply:
> The User who created the issue isn't the one who created the PR (in the general case). In an ideal case, the user who created the issue verifies that the user who created the fix did it correctly, as a double-check.

Good point.
Suggestion:
```
16. An issue MUST be closed in one these ways:
  a. A PR with a proposed solution uses the `Closes: ` git tag. This will close the issue upon merging the PR;
  b. The User who created the issue, closes it after a PR with a proposed solution has been merged;
  c. A Maintainer closes the issue, after a PR with a proposed solution has been merged.
```